src/util/bip143.rs

```diff
@@ -215,6 +219,32 @@ impl<'a> SigHashCache<'a> {
     }
 }
 
+impl<R: DerefMut<Target=Transaction>> SigHashCache<R> {
+    /// When the SigHashCache is initialized with a mutable reference to a transaction instead of a
```

Comment:
only thought I have is if you could move this comment into a real code block in unit test?

Reply:
otherwise no objection

Reply:
Rust is magic. As written that block is run in tests.

Reply:
And you don't have to look around to find this test now. Doctests are wonderful.